### PR TITLE
Populating_W_GPM_for_Circ_Loops

### PR DESCRIPTION
**Pull request by claperle**
May need to adjust for units.

### CODE REVIEW

File: rpd_generator/bdl_structure/bdl_commands/circulation_loop.py [hunk 1]
```diff
@@ -143,6 +143,16 @@ def populate_data_elements(self):
             # None of the data elements for ServiceWaterPiping can be populated from model inputs or outputs
             pass
 
+        # Populate pump_power_per_flow_rate (produces W/GPM, may need to convert to W/L*S for schema)
```
Comment (JacksonJ-KC) [inline rpd_generator/bdl_structure/bdl_commands/circulation_loop.py]: safe to remove note about converting units.

File: rpd_generator/bdl_structure/bdl_commands/circulation_loop.py [hunk 2]
```diff
@@ -143,6 +143,16 @@ def populate_data_elements(self):
...
+        if pump_name is not None:
+            pump_pd = self.rmd.bdl_obj_instances.get(pump_name)
```
Comment (JacksonJ-KC) [inline rpd_generator/bdl_structure/bdl_commands/circulation_loop.py]: pump_pd?

File: rpd_generator/bdl_structure/bdl_commands/pump.py [hunk 3]
```diff
@@ -34,6 +34,7 @@ def populate_data_elements(self):
         self.qty = int(self.try_float(self.keyword_value_pairs.get("NUMBER")))
         requests = self.get_output_requests()
         output_data = self.get_output_data(requests)
```
Comment (JacksonJ-KC) [inline rpd_generator/bdl_structure/bdl_commands/pump.py]: remove this line and replace any references to `output_data` with `self.output_data`

File: rpd_generator/bdl_structure/bdl_commands/circulation_loop.py [hunk 4]
```diff
@@ -143,6 +143,16 @@ def populate_data_elements(self):
...
+            output_data = pump_pd.output_data
+            self.pump_power_per_flow_rate = (
+                output_data["Pump - Power (kW)"]
```
Comment (JacksonJ-KC) [inline rpd_generator/bdl_structure/bdl_commands/circulation_loop.py]: to be safe let's use `output_data.get("Pump - Power (kW)")`  and `output_data.get("Pump - Flow (gal/min)")`

**Review by JacksonJ-KC — APPROVED** [review]
Looks good, just some minor updates before merging

File: rpd_generator/bdl_structure/bdl_commands/circulation_loop.py (hunk 1 above)
Comment (claperle) [reply rpd_generator/bdl_structure/bdl_commands/circulation_loop.py]: Removed

File: rpd_generator/bdl_structure/bdl_commands/circulation_loop.py (hunk 2 above)
Comment (claperle) [reply rpd_generator/bdl_structure/bdl_commands/circulation_loop.py]: @JacksonJ-KC  Seems like I used this based on what I did when creating the hourly schedules code. Not sure what it means. Do you have a suggestion for a better variable name?

File: rpd_generator/bdl_structure/bdl_commands/pump.py (hunk 3 above)
Comment (claperle) [reply rpd_generator/bdl_structure/bdl_commands/pump.py]: I think I need to remove 2 lines right?

File: rpd_generator/bdl_structure/bdl_commands/circulation_loop.py (hunk 2 above)
Comment (JacksonJ-KC) [reply rpd_generator/bdl_structure/bdl_commands/circulation_loop.py]: ohh I see! lets use `loop_pump` instead?

**Pull request merged**